src/ORM/Table.php

```diff
@@ -2174,9 +2174,9 @@ protected function _update(EntityInterface $entity, array $data)
      * any one of the records fails to save due to failed validation or database
      * error.
      *
-     * @param \Cake\Datasource\EntityInterface[]|\Cake\Datasource\ResultSetInterface $entities Entities to save.
+     * @param array<\Cake\Datasource\EntityInterface>|\Cake\Datasource\ResultSetInterface $entities Entities to save.
```

Comment:
This feels like a slippery slope.

Reply:
This is a better way, we couldn't use it earlier because IDEs hadn't caught up with this syntax.

Reply:
It's fine with me but a lot of hints to change if phpstan keeps at it.

I could write a phpcs transform if we need to update more.